### ISSUE TITLE
False positives with Layout/DotPosition

### ISSUE THREAD

**Issue by joeosburn** [rubocop/rubocop#10210]
I am using the `Layout/DotPosition` cop with `trailing` style. I just updated to Rubocop `1.22.2` from `1.22.1`.

I am now getting a whole lot of false positives with this cop whenever there are multiline receivers.

For example:

```
records.map do |record|
  record + 1
end.uniq
```

now gets incorrectly flagged:

```
C: [Correctable] Layout/DotPosition: Place the . on the previous line, together with the method call receiver.
      end.uniq
```
  
## RuboCop version
```
$ bundle exec rubocop -V
1.22.2 (using Parser 3.0.2.0, rubocop-ast 1.12.0, running on ruby 3.0.2 arm64-darwin20)
  - rubocop-performance 1.11.5
```
